packages/server/graphql/mutations/helpers/notifications/notifyMattermost.ts

```diff
@@ -0,0 +1,309 @@
+import {SlackNotificationEventEnum as EventEnum} from '../../../../database/types/SlackNotification'
+import makeAppURL from 'parabol-client/utils/makeAppURL'
+import appOrigin from '../../../../appOrigin'
+import MattermostServerManager, {
+  MattermostApiResponse
+} from '../../../../utils/MattermostServerManager'
+import segmentIo from '../../../../utils/segmentIo'
+import sendToSentry from '../../../../utils/sendToSentry'
+import {DataLoaderWorker} from '../../../graphql'
+import {
+  makeFieldsAttachment,
+  makeHackedFieldButtonValue,
+  makeHackedButtonPairFields
+} from './makeMattermostAttachments'
+import getSummaryText from './getSummaryText'
+import MeetingRetrospective from '../../../../database/types/MeetingRetrospective'
+import MeetingAction from '../../../../database/types/MeetingAction'
+import MeetingPoker from '../../../../database/types/MeetingPoker'
+import findStageById from 'parabol-client/utils/meetings/findStageById'
+import {phaseLabelLookup} from 'parabol-client/utils/meetings/lookups'
+import formatWeekday from 'parabol-client/utils/date/formatWeekday'
+import formatTime from 'parabol-client/utils/date/formatTime'
+import {toEpochSeconds} from '../../../../utils/epochTime'
+
+const okResponse: MattermostApiResponse = {
+  ok: true,
+  status: 0
+}
+
+const getWebhookForUserIdTeamId = async (
+  userId: string,
+  teamId: string,
+  dataLoader: DataLoaderWorker
+) => {
+  const mattermost = await dataLoader.get('mattermostBestAuthByUserIdTeamId').load({userId, teamId})
+  return mattermost?.webhookUrl
+}
+
+const notifyMattermost = async (
+  event: EventEnum,
+  webhookUrl: string,
+  userId: string,
+  teamId: string,
+  textOrAttachmentsArray: string | unknown[],
+  notificationText?: string
+) => {
+  const manager = new MattermostServerManager(webhookUrl)
+  const result = await manager
+    .postMessage(textOrAttachmentsArray, notificationText)
+    .catch((err) => new Error(`Mattermost error: ${err.message}`))
+  if (result instanceof Error) {
+    sendToSentry(result, {userId, tags: {teamId, event, webhookUrl}})
+    return result
+  }
+  if (result.ok) {
+    segmentIo.track({
+      userId,
+      event: 'Mattermost notification sent',
+      properties: {
+        teamId,
+        notificationEvent: event
+      }
+    })
+  }
+
+  return result
+}
+
+export const notifyWebhookConfigUpdated = async (
+  webhookUrl: string,
+  userId: string,
+  teamId: string,
+  dataLoader: DataLoaderWorker
+) => {
+  const {name: teamName} = await dataLoader.get('teams').load(teamId)
+  const message = `Integration webhook configuration updated for team \`${teamName}\``
```

Comment:
VERY nitty- i don't particularly like the team name in a code block. 
to avoid magical parsing, we can just inject a 0-width char in every word.
e.g. "My Team" becomes `M\u200by T\u200beam`

Reply:
Great review comment. Ugh, passing user input to a 3rd-party API should have raised sanitization warning bells for me. I'll just strike it for now. It's not a big add, obvi